### DISCUSSION TITLE
Is there a solution for fully automatic generation of bindings for C++ code/headers?

### DISCUSSION THREAD

**Discussion by womaas** [emscripten-core/emscripten#16046] [General]
Is there a solution for fully automatic generation of bindings for C++ code/headers?

**Comment by sbc100** [comment]
Not that I know of.  I think your choices are currently WebIDL binder and embind, but neither are fully automatic. 
  All the different methods are documented at https://emscripten.org/docs/porting/connecting_cpp_and_javascript/index.html
